addon/resolve-attribute-util.js

```diff
@@ -95,11 +95,13 @@ export function resolveValue(key, value, modelName, store, schema, record, paren
     return resolveReferenceOrReferences(store, record, key, value, reference);
   }
 
-  if (Array.isArray(value)) {
-    return resolveArray(key, value, modelName, store, schema, record, recordData);
-  }
   let nested = computeNestedModel(key, value, modelName, schemaInterface, schema);
   if (nested) {
+    // handle the case where nested returns an array.
+    if (Array.isArray(nested)) {
+      return resolveArray(key, value, modelName, store, schema, record, recordData);
```

Comment:
This doesn't look right to me.  If `computeNestedModel` returns an array, it should be similar to what it returns in the non-array case, eg

```js
[{
  id: 'some-id-or-null',
  type: 'some-type',
  attributes: { / * ... */ },
}, {
  id: 'some-id-or-null',
  type: 'some-type',
  attributes: { / * ... */ },
}]
```

but your call to `resolveArray` is passing `value`.  It's not clear to me how this would work.